include/umf/memory_pool.h

```diff
@@ -137,15 +135,12 @@ umf_memory_pool_handle_t umfPoolByPtr(const void *ptr);
 ///
 /// \brief Retrieve memory providers associated with a given pool.
 /// \param hPool specified memory pool
-/// \param hProviders [out] pointer to an array of memory providers. If numProviders is not equal to or
-///        greater than the real number of providers, UMF_RESULT_ERROR_INVALID_ARGUMENT is returned.
-/// \param numProviders [in] number of memory providers to return
-/// \param numProvidersRet pointer to the actual number of memory providers
+/// \param hProvider [out] memory providers handle.
 /// \return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+///         UMF_RESULT_ERROR_INVALID_ARGUMENT if hProvider is NULL
 enum umf_result_t
-umfPoolGetMemoryProviders(umf_memory_pool_handle_t hPool, size_t numProviders,
-                          umf_memory_provider_handle_t *hProviders,
-                          size_t *numProvidersRet);
+umfPoolGetMemoryProviders(umf_memory_pool_handle_t hPool,
```

Comment:
Should we change the name to `umfPoolGetMemoryProviders`? (without the `s`)

Reply:
Done